stage1/init/init.go

```diff
@@ -230,23 +228,15 @@ func getArgsEnv(p *Pod, flavor string, systemdStage1Version string, debug bool)
 
 	switch flavor {
 	case "coreos":
-		// when running the coreos-derived stage1 with unpatched systemd-nspawn we need some ld-linux hackery
-		args = append(args, filepath.Join(common.Stage1RootfsPath(p.Root), interpBin))
 		args = append(args, filepath.Join(common.Stage1RootfsPath(p.Root), nspawnBin))
 		args = append(args, "--boot") // Launch systemd in the pod
 
-		// Note: the coreos flavor uses systemd-nspawn v215 but machinedRegister()
-		// checks for the nspawn registration method used since v216. So we will
-		// not register when the host has systemd v215.
 		if machinedRegister() {
```

Comment:
machinedRegister() checks that machined on the host supports the API used from nspawn v216+. The comment says that if we had v215 on the host, it would not register since we check v216+ on the host.

I think it is fine to remove the comment, there is already a comment in the function itself about v215 vs v216+.